pkg/component/controller/etcd_member_reconciler.go

```diff
@@ -161,29 +158,25 @@ func (e *EtcdMemberReconciler) Start(ctx context.Context) error {
 			Until(ctx, func(member *etcdv1beta1.EtcdMember) (bool, error) {
 				lastObservedVersion = member.ResourceVersion
 				log.Debugf("watch triggered on %s", member.Name)
-				if e.leaderElector.IsLeader() {
-					if err := e.resync(ctx); err != nil {
-						log.WithError(err).Error("failed to resync etcd members")
-					}
-				} else {
-					log.Debug("Not the leader, skipping")
+				if err := e.resync(ctx, client); err != nil {
+					log.WithError(err).Error("failed to resync etcd members")
 				}
 				// Never stop the watch
 				return false, nil
 			})
-		if err != nil {
-			if errors.Is(err, context.Canceled) {
-				log.WithError(err).Info("watch terminated")
-			} else {
-				log.WithError(err).Error("watch terminated unexpectedly")
-			}
-		}
-	}()
 
-	return nil
+		if canceled := context.Cause(ctx); errors.Is(err, canceled) {
+			log.WithError(err).Info("Watch terminated")
```

Comment:
Nit: Shouldn't watch be lower case?

Reply:
I'm usually using upper case for logs and lower case for errors.

Reply:
Fair enough